core/src/main/java/org/apache/calcite/prepare/CalciteCatalogReader.java

```diff
@@ -138,6 +139,31 @@ public CalciteCatalogReader withSchemaPath(List<String> schemaPath) {
     return config;
   }
 
+  /**
+   * Gets the procedure with the given name if it exists.
+   *
+   * @param names The name of the procedure, list represents possibility of a
+   *              compound identifier
+   *
+   * @return The found procedure, or null
+   */
+  private Procedure getProcedure(List<String> names) {
```

Comment:
This looks identical to `getMacro()`. Can we condense these two cases into a single function?